app/ide-desktop/lib/dashboard/src/utilities/Debug.tsx

```diff
@@ -0,0 +1,60 @@
+/** @file Utilities related to debugging. */
+import * as React from 'react'
```

Comment:
I have no idea why we have this file. What feature of current chrome/react devtools we are missing so do we need to add a component?

Reply:
this provides a convenient way to track changed props for a single element, IME it's quite difficult to filter renders to just the ones that affect a specific component, and even more difficult to find out the props that changed, let alone the values of the changed props

Reply:
There's a flame graph feature: 
![CleanShot 2024-06-16 at 16 46 29](https://github.com/enso-org/enso/assets/61194245/89a95803-9b48-4d35-bd1f-0b893b2bc72a)

Maybe this can help?

Reply:
i do indeed use react devtools, but the flame graph is, in fact, not useful for determining which functions were called - given that the flame graph shows the component tree, not the call stack. and i find the render reasons to be cryptic, and missing the call stack of the original function that triggered the rerender in the first place.

not to mention, the issue here is that the bug was occurring even when *no* renders were happening, so react devtools could not possibly be useful in the first place.

Reply:
Oh, that makes sense. thanks for the clarification.

I still doubt we need this component because we might have forgotten to remove usage from the codebase and debug logs will appear in production ... yikes!

Reply:
should be easy enough to add a lint rule to make it fail CI checks... thoughts?

Reply:
I'm fine with that